Priority Queue/FastPriorityQueue.cs

```diff
@@ -122,40 +122,49 @@ public void Enqueue(T node, float priority)
             _numNodes++;
             _nodes[_numNodes] = node;
             node.QueueIndex = _numNodes;
-            CascadeUp(_nodes[_numNodes]);
+            CascadeUp(node);
         }
 
         #if NET_VERSION_4_5
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         #endif
         private void Swap(T node1, T node2)
         {
+            // Cache to local for faster repeated access
+            int index1 = node1.QueueIndex;
+            int index2 = node2.QueueIndex;
+
             //Swap the nodes
-            _nodes[node1.QueueIndex] = node2;
-            _nodes[node2.QueueIndex] = node1;
+            _nodes[index1] = node2;
+            _nodes[index2] = node1;
 
             //Swap their indicies
-            int temp = node1.QueueIndex;
-            node1.QueueIndex = node2.QueueIndex;
-            node2.QueueIndex = temp;
+            node1.QueueIndex = index2;
+            node2.QueueIndex = index1;
         }
 
         //Performance appears to be slightly better when this is NOT inlined o_O
         private void CascadeUp(T node)
         {
             //aka Heapify-up
-            int parent = node.QueueIndex / 2;
-            while(parent >= 1)
+            int nodeIndex = node.QueueIndex;
+            int parentIndex = nodeIndex >> 1;
```

Comment:
I'd have to look up exactly what those instructions mean, but I expect the extra is to make negative numbers round towards 0 instead of down. Making the index variables all unsigned should therefore get the same improvement. That would be a change visible to users of the queue, though, and break compilation of existing user code.